packages/hydrogen/src/csp/Script.example.jsx

```diff
@@ -23,6 +23,13 @@ export default function App() {
         {/* Note you don't need to pass a nonce to the script component 
         because it's automatically added */}
         <Script src="https://some-custom-script.js" />
+        {/* Nonce is not supported with `waitForHydration`. Instead you
+        need to add the domain of the script directly to your Content 
+        Securitiy Policy directives. */}
```

Comment:
Should we explain why? Like `Nonce is not supported with "waitForHydration" due to browser security` or `For security reasons, nonce is not supported with "waitForHydration"`.